Diagnostics/Diagnostics_utils/PlotMCMC_Diagnosis.cpp

```diff
@@ -88,16 +89,19 @@ void MakePlot(TString fname1, TString fname2,TString fname3, TString fname4)
                     name = dirname + "/" + name;
                     std::cout<<name<<std::endl;
 
-                if (std::string(subkey->GetClassName()) != "TH1D") continue;
+                if (std::string(subkey->GetClassName()) != "TH1D"){continue;}
+				else{std::cout << "continuing along my way" << std::endl;}
 
+				
                     TH1D* blarb[4];
+					std::cout << "Looking for " << name.c_str() << " from file " << fname1.Data() << std::endl; 
                     blarb[0] = (TH1D*)infile->Get(name.c_str())->Clone();
                     //KS: Some fixe pramas can go crazy
                     if(TMath::IsNaN(blarb[0]->GetBinContent(1)) ) continue;
                     
                     //KS: This is unfortunately hardcoded, need to find better way to write this 
                     //blarb[0]->GetListOfFunctions()->ls();
-                    delete blarb[0]->GetListOfFunctions()->FindObject("Fitter");
+                    //delete blarb[0]->GetListOfFunctions()->FindObject("Fitter");
                     blarb[0]->SetLineStyle(kSolid);
```

Comment:
This worked fine in ROOT5 but crashes in ROOT6. We need this to remove TF1 otherwise overlaying several traces is nightmare
Can you add such function and use it everyehre delete blarb[0]->GetListOfFunctions()->FindObject("Fitter"); is being used?
```
//KS: Remove fitted TF1 from hist to make comparison easier
void RemoveFitter(TH1D* hist)
{
  TList *listOfFunctions = hist->GetListOfFunctions();
  TF1 *fitter = dynamic_cast<TF1*>(listOfFunctions->FindObject("Fitter"));

  listOfFunctions->Remove(fitter);
  delete fitter;
}
```